### PR TITLE
Mark Core/Infra Rest API for serverless

### PR DESCRIPTION
**Pull request by pgomulka**
This commit exposes core/infra owned REST APIs for serverless deployment.
It uses @ServerlessScope(Scope.PUBLIC) to achieve this.

follows up on https://github.com/elastic/elasticsearch/pull/94037

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeIndexDiskUsageAction.java [hunk 1]
```diff
@@ -25,6 +27,7 @@
 
 import static org.elasticsearch.rest.RestRequest.Method.POST;
 
+@ServerlessScope(Scope.PUBLIC)
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeIndexDiskUsageAction.java:30]: Is this public or internal?

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeIndexDiskUsageAction.java:30]: as per our discussion during the sync and notes left afterwards it should be public
```
Useful internally as well as for users in combination with field_usage_stats
```

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeIndexDiskUsageAction.java:30]: Ah right

**Review by thecoop — APPROVED** [review]
LGTM

**Pull request merged**